orca-sql/src/main/kotlin/com/netflix/spinnaker/orca/sql/pipeline/persistence/SqlExecutionRepository.kt

```diff
@@ -268,16 +268,30 @@ class SqlExecutionRepository(
     pipelineConfigId: String,
     criteria: ExecutionCriteria
   ): Observable<Execution> {
-    val select = jooq.selectExecutions(
-      PIPELINE,
-      conditions = {
-        it.where(field("config_id").eq(pipelineConfigId))
-          .statusIn(criteria.statuses)
-      },
-      seek = {
-        it.orderBy(field("id").desc()).limit(criteria.pageSize)
-      }
-    )
+    val select = if (criteria.statuses.isEmpty() || criteria.statuses.size == ExecutionStatus.values().size) {
```

Comment:
I wouldn't mind a fast comment here about why this conditional exists. I won't recall in 6 months & would be nice to not git blame dive.